x-pack/plugins/monitoring/server/lib/cluster/get_index_patterns.ts

```diff
@@ -136,3 +136,9 @@ export function getNewIndexPatterns({
   const dsIndexPattern = getDsIndexPattern({ type, moduleType, dataset, namespace, config, ccs });
   return `${legacyIndexPattern},${dsIndexPattern}`;
 }
+
+const getDataset = (product: string) => (metricset: string) => `${product}.stack_monitoring.${metricset}`;
```

Comment:
we could enforce the possible index patterns that could be used here `const getDataset = (product: INDEX_PATTERN_TYPES)`. wdyt?

To make it more consistent with the rest of the functions on this file, could we rename `product` to `moduleType` and `metricset` to `dataset`?

Reply:
Makes sense, I'll send an update